pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5034,6 +5055,35 @@ int BPF_KPROBE(trace_ret_do_splice)
     #endif // CORE && Version < 5.8
 }
 
+SEC("kprobe/load_elf_phdrs")
+int BPF_KPROBE(trace_load_elf_phdrs)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+    if (!should_trace((&data.context)))
+        return 0;
+
+    file_id_t elf = {};
+    struct file *loaded_elf = (struct file*)PT_REGS_PARM2(ctx);
+    const char *elf_pathname = (char *)get_path_str(GET_FIELD_ADDR(loaded_elf->f_path));
+    bpf_probe_read_str(elf.pathname, sizeof(elf.pathname), elf_pathname);
+    elf.device = get_dev_from_file(loaded_elf);
+    elf.inode = get_inode_nr_from_file(loaded_elf);
+
+    bpf_map_update_elem(&interpreter_map, &data.context.host_tid, &elf, BPF_ANY);
```

Comment:
I see that this is the only place that updates the interpreter_map.
If that so, why do we need to send this data in sched_process_exec as well? Isn't it enough to send it here?

Reply:
We want to be able to tell the interpreter in the `sched_process_exec` event.
I can also do it in user-mode, but I think it is preferable to do it in the eBPF program.
WDYT?

Reply:
So for which use do we need an event of load_elf_phdrs in addition to this?

Reply:
I added it in additional for testability reasons.
This way if there is a problem with the `sched_process_exec` output the hook could be checked.
I think it is a best practice to have events for each probe we create so we could check them, maybe even automatically in the future.